src/java/htsjdk/samtools/SamPairUtil.java

```diff
@@ -466,6 +467,7 @@ private void advance() {
         }
 
         public SAMRecord next() {
+            if(!hasNext()) { throw new NoSuchElementException(); }

```

Comment:
hasNext() here seems to throw an exception if the iterator is closed. Is this a reasonable thing to do? I would imagine that a better action would be to return false...not your fault, but an indication of the state of affairs. @alecw I think you added these assertions, do you remember why?